Tasks/TerraformTask/TerraformTaskV4/src/id-token-generator.ts

```diff
@@ -0,0 +1,49 @@
+import tasks = require('azure-pipelines-task-lib/task');
+import { getHandlerFromToken, WebApi } from "azure-devops-node-api";
+import { ITaskApi } from "azure-devops-node-api/TaskApi";
+
+export async function generateIdToken(connectedService : string): Promise<string> {
+    let tokenGenerator = new TokenGenerator();
+    return await tokenGenerator.generate(connectedService);
+}
+
+export interface ITokenGenerator {
+    generate(connectedService : string): Promise<string>;
+}
+
+export class TokenGenerator implements ITokenGenerator {

```

Comment:
Please use this library instead - https://github.com/microsoft/azure-pipelines-tasks-common-packages/blob/main/common-npm-packages/artifacts-common/webapi.ts#L34

Reply:
This is done. I have also updated the other PR to use the same method here: https://github.com/jason-johnson/azure-pipelines-tasks-terraform/pull/354/files#diff-e06f07bc6edfab65173abad59fd62d43f1fb878bc798c52e03325f7a78271a15